api/v1beta1/packetcluster_webhook.go

```diff
@@ -64,17 +81,26 @@ func (c *PacketCluster) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
-	if !reflect.DeepEqual(c.Spec.Facility, old.Spec.Facility) {
+	if !reflect.DeepEqual(c.Spec.VIPManager, old.Spec.VIPManager) {
 		allErrs = append(allErrs,
-			field.Invalid(field.NewPath("spec", "Facility"),
-				c.Spec.Facility, "field is immutable"),
+			field.Invalid(field.NewPath("spec", "VIPManager"),
+				c.Spec.VIPManager, "field is immutable"),
 		)
 	}
 
-	if !reflect.DeepEqual(c.Spec.VIPManager, old.Spec.VIPManager) {
+	// Must have at least Metro or Facility specified
+	if len(c.Spec.Facility) == 0 && len(c.Spec.Metro) == 0 {
 		allErrs = append(allErrs,
-			field.Invalid(field.NewPath("spec", "VIPManager"),
-				c.Spec.VIPManager, "field is immutable"),
+			field.Invalid(field.NewPath("spec", "Metro"),
+				c.Spec.Metro, "Metro is required when Facility is not set"),
+		)
+	}
+
+	// Must have only one of Metro or Facility
+	if len(c.Spec.Facility) > 0 && len(c.Spec.Metro) > 0 {
+		allErrs = append(allErrs,
+			field.Invalid(field.NewPath("spec", "Facility"),
+				c.Spec.Facility, "Metro and Facility are mutually exclusive"),
```

Comment:
How verbose can we be in these errors?  I think it'd be preferable to steer a user who provided both fields towards removing facility rather than removing metro.

Reply:
I've resolved all the above.